rl_games/algos_torch/SACAgent.py

```diff
@@ -0,0 +1,458 @@
+        self.config = config
+        self.env_config = config.get('env_config', {})
+        self.num_actors = config.get('num_actors', 1)
+        self.env_name = config['env_name']
+        print("Env name:", self.env_name)
+
+        self.env_info = config.get('env_info')
+        if self.env_info is None:
+            self.vec_env = vecenv.create_vec_env(self.env_name, self.num_actors, **self.env_config)
+            self.env_info = self.vec_env.get_env_info()
+
+        self.sac_device = config.get('device', 'cuda:0')
+        print('Env info:')
+        print(self.env_info)
+
+        self.observation_space = self.env_info['observation_space']
+        self.weight_decay = config.get('weight_decay', 0.0)
+        self.use_action_masks = config.get('use_action_masks', False)
+        self.is_train = config.get('is_train', True)
+
+        self.central_value_config = self.config.get('central_value_config', None)
+        self.has_central_value = self.central_value_config is not None
+
+        if self.has_central_value:
+            self.state_space = self.env_info.get('state_space', None)
+            self.state_shape = None
+            if self.state_space.shape != None:
+                self.state_shape = self.state_space.shape
+
+        self.self_play_config = self.config.get('self_play_config', None)
+        self.has_self_play_config = self.self_play_config is not None
+
+        self.self_play = config.get('self_play', False)
+        self.save_freq = config.get('save_frequency', 0)
+        self.save_best_after = config.get('save_best_after', 100)
+        self.print_stats = config.get('print_stats', True)
+        self.rnn_states = None
+        self.name = base_name
+
+        self.max_epochs = self.config.get('max_epochs', 1e6)
+
+        self.network = config['network']
+        self.rewards_shaper = config['reward_shaper']
+        self.num_agents = self.env_info.get('agents', 1)
+        self.obs_shape = self.observation_space.shape
+
+
+        self.games_to_track = self.config.get('games_to_track', 100)
+        self.game_rewards = torch_ext.AverageMeter(1, self.games_to_track).to(self.sac_device)
+        self.game_lengths = torch_ext.AverageMeter(1, self.games_to_track).to(self.sac_device)
+        self.obs = None
+        
+
+        self.frame = 0
+        self.update_time = 0
+        self.last_mean_rewards = -100500
+        self.play_time = 0
+        self.epoch_num = 0
+        
+        # self.writer = SummaryWriter('ant_runs/' + config['name'] + datetime.now().strftime("_%d-%H-%M-%S"))
+        self.writer = SummaryWriter('tested_new/' + 'ant')
+        # self.writer = SummaryWriter('walker/'+'fixed_buffer')
+        print("Run Directory:", config['name'] + datetime.now().strftime("_%d-%H-%M-%S"))
+
+        self.is_tensor_obses = None
+
+        self.curiosity_config = self.config.get('rnd_config', None)
+        self.has_curiosity = self.curiosity_config is not None
+        if self.has_curiosity:
```

Comment:
I removed curiosity support because I've found that rnd curiosity works the best only for some atari games. But exploration based reward tuned for special case performs much better.